src/movies.js

```diff
@@ -1,14 +1,151 @@
+      "year": 1957,
+      "director": "Sidney Lumet",
+      "duration": "1h 36min",
+      "genre": [
+        "Crime",
+        "Drama"
+      ],
+      "rate": 8.9
+    }
+            ]; */
+
+
+function orderByYear(filmArr) {
+    const copyFilms = [...filmArr];
+    const sorted = copyFilms.sort( function(filmA, filmB) {
+        if (filmA.year < filmB.year) {
+            return -1;
+        }
+        else if (filmA.year > filmB.year) {
+            return 1;
+        }
+        // and if the year is same:
+        // else block would start here but i dont wanna use it if 4 if-statements work fine :D
+        if ( filmA.title.toLowerCase() > filmB.title.toLowerCase() ) { // yes??? yes.
+            return 1; // 
+        }
+        else if ( filmA.title.toLowerCase() < filmB.title.toLowerCase() ) {
+            return -1;
+        }
+
+    } );
+    return sorted; //
+}
+//films = orderByYear(movies);
+//console.log(films);
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
+//One of the most famous directors in cinema is Steven Spielberg, and he has some really awesome drama movies 
+//that are on our list, but we want to know how many of them are there.
+//Go ahead and create a howManyMovies() function that receives an array as a parameter and filter the array 
+//so we can have only the drama movies where Steven Spielberg is the director.
+function howManyMovies(filmArr) 
+{
+    const stevens = filmArr.filter( film => film.director === "Steven Spielberg" ) // director: 'Steven Spielberg'
+    return stevens.length; // why limit yourself to just drama ?
+}
+
 
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
+//Create a orderAlphabetically() function, that receives an array and returns an array of first 20 titles, alphabetically ordered. 
+//Return only the title of each movie, and 
+//if the array you receive has less than 20 movies, return all of them.
+
+function orderAlphabetically(filmArr) {
+
+    const mutableFilms = [...filmArr]; // making a copy of films list that i'm allowed to mutate
+    const titles = [];
+    mutableFilms.forEach( (film) => titles.push(film.title) )
+    orderedTitles = titles.sort();
+    return orderedTitles.splice(0,20);
+}
+
+console.log(orderAlphabetically(movies));
 
 // Iteration 4: All rates average - Get the average of all rates with 2 decimals
+//Create a ratesAverage() function that receives an array as a parameter and solves the challenge.
+//display it on the console. The rate must be returned rounded to 2 decimals!
+
+function ratesAverage(filmArr) {
+    const ratings = [];
+    filmArr.forEach( (film) => ratings.push(film.rate) ) // rip out all the ratings and put them in 1 array
+    //console.log(ratings);
+    avg = (ratings.reduce( (totalRatings, nextRating) => totalRatings + nextRating )) / (ratings.length); // sum all ratings with reduce() to calculate average
+    //console.log(avg); 
+    return avg;
```

Comment:
You need to round the avg to 2 decimals and make sure it's a number, in order to pass all tests